inception-project/inception
# #825 External Search Sidebar

#825  External Search Sidebar

**What's in the PR**
* Add new External Search Sidebar to the Annotation Page
* Clicking on the search result imports/opens selected document in annotation page
* Sidebar state persistence
* Highlighting keywords will be in a separate PR.

**How to test manually**
1. Test external search
1.1. Open the app, select a project and select Annotation from the project dashboard.
1.2. Select any document to open Annotation page.
1.3. Confirm that there is a external search icon in the sidebar menu.
1.4. Click on the external search icon and the external search sidebar should open.
1.5. Inside the external search sidebar, select a repository and type in a keyword to search. Results should appear below the search form. Go through the list and jump through pages to check the results.
2. Test importing and opening document
2.1. Click on an unimported row's title link. The document should open on the right, and the unimported tag should be now changed to imported.
2.2. Click on an imported row's title link. The document should open on the right.
2.3. Annotate the document, jump to another document and come back again to see annotations are saved.
3. Test persistence
3.1 In the result list, go to a page other than page 1.
3.2 Click on another item in the sidebar menu and then come back again. 
3.3 Check that previously selected repository, search keyword, results and page are unchanged.

Screenshot
<img width="1421" alt="screen shot 2019-01-17 at 1 08 19 pm" src="https://user-images.githubusercontent.com/636361/51349014-10eeae80-1a59-11e9-8c55-7fb29afbb29f.png">
